core/src/main/java/org/mule/runtime/core/source/scheduler/DefaultSchedulerMessageSource.java

```diff
@@ -85,18 +88,31 @@ public String getPollingUniqueName() {
   }
 
   @Override
-  public void stop() throws MuleException {
+  public synchronized void stop() throws MuleException {
+    if (!started) {
+      return;
+    }
     // Stop the scheduler to address the case when the flow is stop but not the application
     if (schedulingJob != null) {
       schedulingJob.cancel(false);
       schedulingJob = null;
     }
+    this.started = false;
+  }
+
+  public void trigger() {
```

Comment:
Can you explain in javadoc the difference between run() and trigger() given both are public.